File: src/main/java/FlyAway/reservation/dto/ReservationWithoutFlightDto.java
package FlyAway.reservation.dto;

import FlyAway.client.dto.ClientDto;
import FlyAway.flight.aircraft.CabinClass;
import FlyAway.reservation.ReservationStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record ReservationWithoutFlightDto(
        LocalDateTime reservationDate,
        BigDecimal price,
        Integer seatNumber,
        CabinClass cabinClass,
        ReservationStatus status,
        ClientDto clientDto
) {
}

File: frontend/src/app/services/models/reservation-without-flight-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ClientDto } from '../models/client-dto';
export interface ReservationWithoutFlightDto {
  cabinClass?: 'FIRST' | 'BUSINESS' | 'ECONOMY';
  clientDto?: ClientDto;
  price?: number;
  reservationDate?: string;
  seatNumber?: number;
  status?: 'PENDING' | 'ACTIVE' | 'CANCELLED' | 'EXPIRED' | 'COMPLETED' | 'FAILED';
}
